lib/judges/baza.rb

```diff
@@ -219,6 +210,66 @@ def headers
     }
   end
 
+  def request_options_for_push(data, meta)
+    { connecttimeout: @timeout, timeout: @timeout }
+      .then { |req_opts| put_headers(req_opts) }
+      .then { |req_opts| put_body(req_opts, data) }
+      .then { |req_opts| put_meta(req_opts, meta) }
+  end
+
+  def put_headers(req_opts)
+    req_opts.merge(headers:)
+  end
+
+  def put_body(req_opts, data)
+    if @enable_compression
+      put_body_with_compression(req_opts, data)
+    else
+      put_body_without_compression(req_opts, data)
+    end
+  end
+
+  def put_body_with_compression(req_opts, data)
+    body = compress(data)
+    headers = req_opts.fetch(:headers).merge(
+      'Content-Type' => 'application/zip',
+      'Content-Encoding' => 'gzip',
+      'Content-Length' => body.size
+    )
+    req_opts.merge(headers:, body:)
+  end
+
+  def put_body_without_compression(req_opts, body)
+    headers = req_opts.fetch(:headers).merge(
+      'Content-Type' => 'application/octet-stream',
+      'Content-Length' => body.size
+    )
+    req_opts.merge(headers:, body:)
+  end
+
+  def put_meta(req_opts, meta)
+    if meta.any?
+      headers = put_meta_to_headers(req_opts.fetch(:headers), meta)
+      req_opts.merge(headers:)
+    else
+      req_opts
+    end
+  end
+
+  def put_meta_to_headers(headers, meta)
+    value = meta.map { |v| Base64.encode64(v).gsub("\n", '') }.join(' ')
+    headers.merge('X-Zerocracy-Meta' => value)
+  end
+
+  def compress(data)
```

Comment:
@tank-bohr check this: https://www.yegor256.com/2018/08/22/builders-and-manipulators.html